### PR TITLE
Issue 1079 remove login logout lti

### LINKED ISSUE #1079: Disable logout URL when LTI is configured
If you're launching via LTI we should disable the logout link in the UI as it really won't do anything. Maybe should just pass something to the myla globals whether or not to show the logout? This was kind of mentioned on this ticket.

However it's possible to ENABLE_BACKEND_LOGIN and login that way. In that case we should still have the logout link. So this might not be completely straightforward to fix.

### PR DESCRIPTION
**Pull request by pushyamig**
Fixes #1079

This issues removes the logout URL from the UI doesn't send in the Login URL when LTI. But this should be available with SAML/local 

This PR is based off of #1079 1073

### CODE REVIEW

**Comment by jonespm** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ITS_Teaching_And_Learning/my-learning-analytics/pullRequest?prid=6160227&bid=20123248)

**Comment by jonespm** [comment]
I'll review this after it's rebased.

**Comment by pushyamig** [comment]
OK, I rebased. Please check again!

File: dashboard/common/utils.py [hunk 1]
```diff
@@ -86,9 +86,9 @@ def get_myla_globals(request):
         username = current_user.get_username()
         user_courses_info = get_user_courses_info(username, course_id)
 
-    if settings.LOGIN_URL:
+    if not settings.STUDENT_DASHBOARD_LTI:
         login_url = settings.LOGIN_URL
-    if settings.LOGOUT_URL:
+    if not settings.STUDENT_DASHBOARD_LTI:
         logout_url = settings.LOGOUT_URL
```
Comment (ssciolla) [inline dashboard/common/utils.py:92]: For some reason, when I try to access `localhost:5001`, it's doing that looping thing which I've pointed out before: it keeps trying and failing to load. This isn't happening on `master` (you just get a 404 when redirected to `accounts/login`). I can't explain this yet. `myla_globals['login_url']` must be in use somewhere and the logic you changed is breaking something?

**Review by ssciolla — COMMENTED** [review]
This isn't working for me yet. See the comment about going straight to the MyLA domain. This happens whether `ENABLE_BACKEND_LOGIN` is `true` or `false`, I believe.

File: dashboard/common/utils.py (hunk 1 above)
Comment (ssciolla) [reply dashboard/common/utils.py:92]: Just a note to say that this is successfully hiding the "Logout" button, but the behavior around visiting the domain and using the default backend seems to have been disrupted somehow.

**Comment by jonespm** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- assets/src/containers/App.js  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ITS_Teaching_And_Learning/my-learning-analytics/pullRequest?prid=6160227&bid=20123248)

File: assets/src/containers/App.js [hunk 2]
```diff
@@ -4,11 +4,15 @@ import { matchPath } from 'react-router'
...
 import CourseList from './CourseList'
 import Course from './Course'
+import WarningBanner from '../components/WarningBanner'
 
 function App (props) {
   const { location, user, gaId } = props
 
   if (!user.isLoggedIn) {
+    if (user.loginURL === '') {
+      return (<WarningBanner>You should launch this tool from your Canvas Course</WarningBanner>)
```
Comment (ssciolla) [inline assets/src/containers/App.js:14]:
```suggestion
      return (<WarningBanner>This tool needs to be launched from a Canvas course.</WarningBanner>)
```

**Review by ssciolla — APPROVED** [review]
I think this is okay. I'd rather use `null` over the empty string as the default, but I won't insist on that change right now.

It's a little confusing when you use the default login while also using LTI; the "Logout" button is gone from the Avatar drop-down, even when you aren't in the Canvas iframe. Maybe that's alright since it mimics what the user would see? In production, we wouldn't want these on at the same time yet. I think there's probably a way to put in `mylaGlobals` where this is getting launched from (determined by the referrer?) and use that to decide whether to show "Logout". That might make things more consistent, which might be desirable if you wanted to enable a prod backdoor. It'd be nice to hear @jonespm's feedback on this, too.

**Comment by pushyamig** [comment]
> I think there's probably a way to put in mylaGlobals where this is getting launched from (determined by the referrer?) and use that to decide whether to show "Logout". That might make things more consistent

Even we put the logout but it won't do much since the `/accounts/logout` is disabled in Backend. I am not how feasible option based on the referrer you enable the backed and display option. Not for the PR :)

**Comment by pushyamig** [comment]
> It's a little confusing when you use the default login while also using LTI; the "Logout" button is gone from the Avatar drop-down, even when you aren't in the Canvas iframe. Maybe that's alright since it mimics what the user would see?

Yes, When authentication is done in LTI and not through Django form/SAML, then it make sense not to show the logout button.

**Comment by ssciolla** [comment]
> > I think there's probably a way to put in mylaGlobals where this is getting launched from (determined by the referrer?) and use that to decide whether to show "Logout". That might make things more consistent
> 
> Even we put the logout but it won't do much since the `/accounts/logout` is disabled in Backend. I am not how feasible option based on the referrer you enable the backed and display option. Not for the PR :)

If `ENABLE_BACKEND_LOGIN` is `true`, then `/accounts/logout` is still active, isn't it?

**Comment by pushyamig** [comment]
> If ENABLE_BACKEND_LOGIN is true, then /accounts/logout is still active, isn't it?

Yes, here is the logic for it
https://github.com/tl-its-umich-edu/my-learning-analytics/blob/master/dashboard/urls.py#L100

**Comment by ssciolla** [comment]
That's my point You **can** log out if you log in using the `ENABLE_BACKEND_LOGIN` while LTI is also enabled, but it's not visible in the drop-down, because you're always disabling it when LTI is eanbled.

> > If ENABLE_BACKEND_LOGIN is true, then /accounts/logout is still active, isn't it?
> 
> Yes, here is the logic for it
> https://github.com/tl-its-umich-edu/my-learning-analytics/blob/master/dashboard/urls.py#L100

That's my point. You **can** log out if you log in using the backend triggered by `ENABLE_BACKEND_LOGIN` while LTI is also enabled, but it's not visible in the drop-down, because the change you added always hides it when LTI is enabled.

**Comment by pushyamig** [comment]
> That's my point. You can log out if you log in using the backend triggered by ENABLE_BACKEND_LOGIN while LTI is also enabled, but it's not visible in the drop-down, because the change you added always hides it when LTI is enabled.

With LTI, the option is set to `ENABLE_BACKEND_LOGIN = False` and the account URL are disabled. SO you cannot login/logout by typing in URL

**Comment by pushyamig** [comment]
```
CHECK_ENABLE_BACKEND_LOGIN = False if STUDENT_DASHBOARD_SAML or STUDENT_DASHBOARD_LTI else True 

# Allow for ENABLE_BACKEND_LOGIN override
ENABLE_BACKEND_LOGIN = ENV.get("ENABLE_BACKEND_LOGIN", CHECK_ENABLE_BACKEND_LOGIN)
```

**Comment by ssciolla** [comment]
> With LTI, the option is set to `ENABLE_BACKEND_LOGIN = False` and the account URL are disabled. SO you cannot login/logout by typing in URL

That's wrong; you can force it to enable both LTI and the default backend by using the override, specifying `"ENABLE_BACKEND_LOGIN": true` in `env.json`.

**Comment by pushyamig** [comment]
> That's wrong; you can force it to enable both LTI and the default backend by using the override, specifying "ENABLE_BACKEND_LOGIN": true in env.json.

we are enabling Modelbackend with LTI
I felt we are going to set it `ENABLE_BACKEND_LOGIN` = false for LTI and my assumption is we are going to disable the account URL's as well @jonespm  please share your thoughts!

**Comment by jonespm** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- assets/src/containers/App.js  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ITS_Teaching_And_Learning/my-learning-analytics/pullRequest?prid=6160227&bid=20123248)

**Comment by pushyamig** [comment]
@ssciolla upon discussion with @jonespm 
for LTI `ENABLE_BACKEND_LOGIN` will be `False` and `account/login` and `account/logout` will be disabled in backend and not show in the UI.

In case of SAML and Django default forms those will be enabled. 

I feel this PR is serving the purpose as intended

**Review by jonespm — APPROVED** [review]

**Pull request merged**